### PR TITLE
Core - Message Layout Update

### LINKED ISSUE #318: Core - message layout update
## Description

Message layout changes
- move `NetworkID` into transaction essence
- Add `Inputs Commitment` in the transaction essence
- Add `Protocol Version ` to the message

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- fixed #318 
- changed JSON names for address to `pubKeyHash`, `aliasId` or `nftId`, depending on an address type
- fixed send tagged data and simple transaction message

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Unit tests were updated with all newly introduced changes.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/json_parser/payloads.c [hunk 1]
```diff
@@ -236,6 +275,19 @@ int json_transaction_deserialize(cJSON* payload, transaction_payload_t* tx) {
...
+    if (json_get_string(essence_obj, JSON_KEY_INPUTS_COMMITMENT, (char*)inputs_commitment_hex,
+                        sizeof(inputs_commitment_hex)) != JSON_OK) {
+      printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_INPUTS_COMMITMENT);
+      return -1;
+    }
+    if (hex_2_bin(inputs_commitment_hex, sizeof(inputs_commitment_hex), tx->essence->inputs_commitment,
+                  sizeof(tx->essence->inputs_commitment)) != 0) {
+      printf("[%s:%d] can not convert hex to bin number\n", __func__, __LINE__);
+      return -1;
+    }
```
Comment (oopsmonk) [inline src/client/api/json_parser/payloads.c]: can use `json_get_hex_str_to_bin` instead.

File: src/core/models/payloads/transaction.c [hunk 2]
```diff
@@ -12,7 +12,9 @@ transaction_essence_t* tx_essence_new() {
   transaction_essence_t* es = malloc(sizeof(transaction_essence_t));
   if (es) {
     es->tx_type = TRANSACTION_PAYLOAD_ESSENCE;  // 0 to denote a transaction essence.
+    es->network_id = 0;
```
Comment (oopsmonk) [inline src/core/models/payloads/transaction.c]: network id is mandatory and known before creating the transaction, it should be passed by `tx_essence_new` 
`transaction_essence_t* tx_essence_new(uint64_t network_id)`

the same with the protocol version in the message
` core_message_t* core_message_new(uint8_t ver)`

File: src/client/api/json_parser/payloads.c (hunk 1 above)
Comment (matjazv) [reply src/client/api/json_parser/payloads.c]: Done. Fixed also in some other places where it was appropriate to use it.

File: src/core/models/payloads/transaction.c (hunk 2 above)
Comment (matjazv) [reply src/core/models/payloads/transaction.c]: Added `network Id` and `protocol version` as functions parameters.

File: src/client/api/json_parser/common.c [hunk 3]
```diff
@@ -28,39 +28,29 @@ int json_parser_common_address_deserialize(cJSON *json_obj, char const *const js
   char address_hex[ADDRESS_MAX_HEX_BYTES];
```
Comment (rebinmannady93) [inline src/client/api/json_parser/common.c]: address_hex is not used anymore. We can remove it.

**Review by rebinmannady93 — CHANGES_REQUESTED** [review]

File: src/client/api/json_parser/common.c (hunk 3 above)
Comment (matjazv) [reply src/client/api/json_parser/common.c]: Thanks to fetched this.

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**